src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java

```diff
@@ -312,4 +382,109 @@ private int getLeafCountForTheCourse(String rootOrgId, String courseId, String u
+			return strBuilder.toString();
+		}
+
+		List<String> missingAttrib = new ArrayList<String>();
+		if (!requestBody.containsKey(Constants.COURSE_ID)) {
+			missingAttrib.add(Constants.COURSE_ID);
+		}
+
+		if (!requestBody.containsKey(Constants.BATCH_ID)) {
+			missingAttrib.add(Constants.BATCH_ID);
+		}
+
+		if (!requestBody.containsKey(Constants.LIMIT)) {
+			missingAttrib.add(Constants.LIMIT);
+		}
+
+		if (!requestBody.containsKey(Constants.OFFSET)) {
+			missingAttrib.add(Constants.OFFSET);
+		}
+
+		if (missingAttrib.size() > 0) {
+			strBuilder.append("The following parameter(s) are missing. Missing params - [")
+					.append(missingAttrib.toString()).append("]");
+		}
+
+		return strBuilder.toString();
+	}
+
+	public Map<String,Object> getBatchParticipantsByPage(Map<String, Object> request) {
+		Map<String,Object> responseMap = new HashMap<>();
+		Map<String, Object> queryMap = new HashMap<>();
+		queryMap.put(Constants.BATCH_ID, (String) request.get(Constants.BATCH_ID));
+		Map<String, Object> result = new HashMap<String, Object>();
+		List<String> userList = new ArrayList<String>();
+		Boolean active = (Boolean) request.get(Constants.ACTIVE);
+		if (null == active) {
+			active = true;
+		}
+		Integer limit = (Integer) request.get(Constants.LIMIT);
+		if (limit == null) {
+			limit = 10; // check default limit add it as constants
+		}
+		Integer currentOffSetFromRequest = (Integer) request.get(Constants.OFFSET);
+		if (currentOffSetFromRequest == null) {
+			currentOffSetFromRequest = 0;
+		}
+		String pageId = (String) request.get(Constants.PAGE_ID);
+		String previousPageId = null;
+		int currentOffSet = 0;
+		String currentPagingState = null;
+		Long count = cassandraOperation.getCountOfRecordByIdentifier(Constants.KEYSPACE_SUNBIRD_COURSES,
+				Constants.TABLE_ENROLLMENT_BATCH_LOOKUP, queryMap, Constants.USER_ID);
+		do {
+			Map<String,Object> response = cassandraOperation.getRecordByIdentifierWithPage(Constants.KEYSPACE_SUNBIRD_COURSES,
+					Constants.TABLE_ENROLLMENT_BATCH_LOOKUP, queryMap,
+					null, pageId, (Integer) request.get(Constants.LIMIT));
+			currentPagingState = (String) response.get(Constants.PAGE_ID);
+			if (org.codehaus.plexus.util.StringUtils.isBlank(previousPageId) && org.codehaus.plexus.util.StringUtils.isNotBlank(currentPagingState)) {
+				previousPageId = currentPagingState;
+			}
+			pageId = currentPagingState;
+			List<Map<String, Object>> userCoursesList = (List<Map<String, Object>>) response.get(Constants.RESPONSE);
+			if (org.apache.commons.collections.CollectionUtils.isEmpty(userCoursesList)) {
+				//Set null so that client knows there are no data to read further.
+				previousPageId = null;
+				break;
+			}
+			for (Map<String, Object> userCourse : userCoursesList) {
+				//From this page, we have already read some records, so skip the records
+				if (currentOffSetFromRequest > 0) {
+					currentOffSetFromRequest--;
+					continue;
+				}
+				if (userCourse.get(Constants.ACTIVE) != null
```

Comment:
Do add an else part for this condition... 
inside else - just log an error statement with userId, courseId and batchId details...